### PR TITLE
Ch data validation #162268600

### PR DESCRIPTION
**Pull request by niomwungeri-fabrice**
#### What does this PR do?
Add more data validation on parcel and user objects
#### Description of Task to be completed?
Cover sanity checks:
- Invalid email
- invalid token
- unauthorized access

#### How should this be manually tested?
Using Postman:
- Enter invalid token when signup/sign in
- Access protected route without valid 
- Access protected route without jwt token

#### What are the relevant pivotal tracker stories?
162268600

### CODE REVIEW

File: src/controller/ParcelController.js [hunk 1]
```diff
@@ -79,8 +83,13 @@ const Parcels = {
...
     try {
       const { rows } = await db.query(findOneQuery, [req.params.parcelId, req.user.id]);
+      if (rows[0]) {
+        if (rows[0].status.toUpperCase() === 'delivered'.toUpperCase() || rows[0].status.toUpperCase() === 'Cancelled'.toUpperCase()) {
+          return res.status(400).send({ message: 'Ooops, The parcel has been delived or concelled already, Cancel denied!', status: 400 });
+        }
+      }
       if (!rows[0]) {
-        return res.status(404).send({ message: 'parcel not found', status: 404 });
+        return res.status(400).send({ message: 'parcel not found', status: 400 });
```
Comment (rotimi-babalola) [inline src/controller/ParcelController.js]: This should be a `404`

File: src/controller/ParcelController.js [hunk 2]
```diff
@@ -105,7 +114,7 @@ const Parcels = {
     try {
       const { rows } = await db.query(findOneQuery, [req.params.parcelId, req.user.id]);
       if (!rows[0]) {
-        return res.status(404).send({ message: 'Parcel not found', status: 404 });
+        return res.status(400).send({ message: 'Parcel not found', status: 400 });
```
Comment (rotimi-babalola) [inline src/controller/ParcelController.js]: Same as 👆🏽

File: src/controller/ParcelController.js [hunk 3]
```diff
@@ -129,7 +138,7 @@ const Parcels = {
...
```
Comment (rotimi-babalola) [inline src/controller/ParcelController.js]: Same as 👆🏽

File: src/controller/ParcelController.js [hunk 4]
```diff
@@ -153,7 +162,7 @@ const Parcels = {
...
```
Comment (rotimi-babalola) [inline src/controller/ParcelController.js]: Same as 👆🏽

File: src/controller/UserController.js [hunk 5]
```diff
@@ -20,6 +20,9 @@ const Users = {
         message: 'Account Created Successfully', status: 201, data: rows[0], token,
       });
     } catch (error) {
+      if (error.routine === '_bt_check_unique') {
+        return res.status(400).send({ message: `User with this email '${req.body.email}' already exist`, status: 409 });
```
Comment (rotimi-babalola) [inline src/controller/UserController.js]: Why is the status sent in the response different from the status in the sent in the JSON object?

File: src/controller/UserController.js [hunk 6]
```diff
@@ -28,6 +31,9 @@ const Users = {
     const text = 'SELECT * FROM users WHERE email = $1';
     try {
       const { rows } = await db.query(text, [req.body.email]);
+      if (!rows[0]) {
+        return res.status(400).send({ message: 'User does not exist' });
```
Comment (rotimi-babalola) [inline src/controller/UserController.js]: The error code should be `404`

**Review by rotimi-babalola — CHANGES_REQUESTED** [review]
Your PRs should be specific to one task only. I see the name of this PR is data validation but you have changes in your UI in this PR. The changes to your UI should be in a separate branch which you should merge in a different PR (you can merge this for now but note this in the future).

Also fix the other comments I left

File: src/controller/UserController.js (hunk 5 above)
Comment (niomwungeri-fabrice) [reply src/controller/UserController.js]: I learned about 409 status recently and changed it from returned JSON but I forgot to change it in the response body. will change now.

**Pull request merged**